osm2lanes/src/lib.rs

```diff
@@ -271,6 +272,8 @@ pub struct LaneSpec {
     /// (though local rules might still dictate restrictions).
     /// Turns for specific vehicle types (`turn:bus:lanes` and such) are not yet captured.
     pub allowed_turns: EnumSet<TurnDirection>,
+
+    pub lane: Option<Lane>,
```

Comment:
I like this as a way to plumb the more detailed representation along. `LaneSpec` here probably needs lots of work (direction and lane type particularly), but new features there have to be understood by downstream consumers to be useful -- at minimum, the street explorer. Having the more detailed `Lane` could be a way to update those downstream consumers a bit gradually.

Reply:
>direction and lane type particularly

I agree. I just didn't touch them for now as you mentioned in another issue that direction especially changing would break a lot of the road placement and routing code, so i left them as is for now.